Source/Private/ASLayoutTransition.mm

```diff
@@ -125,7 +125,12 @@ - (void)applySubnodeRemovals
 
   ASDisplayNodeLogEvent(_node, @"removeSubnodes: %@", _removedSubnodes);
   for (ASDisplayNode *subnode in _removedSubnodes) {
-    [subnode _removeFromSupernode];
+    // We have to check in this case if the supernode of the subnode is indeed the _node that executes a layout transition.
+    // It can happen that a node already did a layout transition and added this subnode, in this case the subnode would
+    // would be removed from the new node instead of _node
+    if (subnode.supernode == _node) {
```

Comment:
The instance lock of subnode is acquired twice in this method: inside supernode getter and then inside `_removeFromSupernode`. Would it be possible for supernode to be changed during the short time gap before the 2 lock acquisitions? Should we add a new `_removeFromSupernodeIfEqualTo:(ASDisplayNode *)supernode` method? Inside that method we can just return if `_supernode != supernode`?